core/src/main/scala/shapeless/poly.scala

```diff
@@ -113,6 +113,59 @@ object PolyDefns extends Cases {
       }
   }
 
+  final case class BindFirst[F, Head](head: Head) extends Poly
+
+  object BindFirst {
+    implicit def bindFirstCase[BF, F, Head, Tail <: HList, Result0](
+        implicit
+        unpack2: BF <:< BindFirst[F, Head],
+        witnessBF: Witness.Aux[BF],
+        finalCall: Case.Aux[F, Head :: Tail, Result0]
+    ): Case.Aux[BF, Tail, Result0] = new Case[BF, Tail] {
+      type Result = Result0
+      val value: Tail => Result = { tail: Tail =>
+        finalCall.value(witnessBF.value.head :: tail)
+      }
+    }
+  }
+
+  final case class Curried[F, ParameterAccumulator <: HList](parameters: ParameterAccumulator) extends Poly1
```

Comment:
@Atry do you remember what the `ParameterAccumulator` type parameter is for?
It seems unused to me = always `HNil`

Reply:
```
    object p extends Poly3 {
      implicit def x = at[Int, Double, String] { (i, d, s) =>
        s"$i, $d, $s"
      }
    }

    val c = Poly.curried(p)
    val c1 = c(1)
    val c2 = c1(42.5)
```

Try printing `c2`.

Reply:
Ah ok that makes sense :+1: